lldb/tools/lldb-dap/lldb-dap.cpp

```diff
@@ -4020,6 +4020,10 @@ void request_variables(DAP &dap, const llvm::json::Object &request) {
                           std::optional<std::string> custom_name = {}) {
         if (!child.IsValid())
           return;
+        if (child.IsSynthetic() && (child.GetType().IsPointerType() || child.GetType().IsReferenceType())) {
```

Comment:
@clayborg , what do you think about this? This pretty much matches the experience from CLI, i.e. the variable name is prepended with `*`, but the synthetic value is displayed right away.